pytorch_lightning/utilities/cli.py

```diff
@@ -690,12 +700,31 @@ def _parser(self, subcommand: Optional[str]) -> LightningArgumentParser:
         action_subcommand = self.parser._subcommands_action
         return action_subcommand._name_parser_map[subcommand]
 
-    def _add_configure_optimizers_method_to_model(self, subcommand: Optional[str]) -> None:
-        """Adds to the model an automatically generated ``configure_optimizers`` method.
+    @staticmethod
+    def configure_optimizers(
```

Comment:
Quick question. Do you believe we could support multiple optimizers, schedulers?

I was thinking about the following idea.
Assuming the following command line:

```py
--optimizer_1=Adam
--optimizer_2=.SGD
--lr_scheduler=Adam
```

and the following configure_optimizers method

```
def configure_optimizers(self):
     opt_gen = ...
     opt_dis = ...
     scheduler = ...(opt_gen)
     return [opt_gen, opt_dis], scheduler
```

We could call the real `configure_optimizers`, get the 2 optimizers and scheduler, and use those to make the same parameters & scheduler - optimizer association based on the command line.

Just an idea.

Reply:
One of the features of the CLI is removing the need for implementing `configure_optimizers` on the model for basic uses.

What you suggest goes against that, as the model implementation controls the CLI behavior.

This PR already makes the CLI behavior overridable. It doesn't support multiple optimizers at the moment but it could be supported.